src/gbt_model_builder.pyx

```diff
@@ -179,3 +181,110 @@ def gbt_reg_model_builder(n_features, n_iterations):
+        self.daal_model_ = get_gbt_model_from_xgboost(booster, xgb_params)
+        self._get_params_from_xgboost(xgb_params)
+
+    def _build_model_from_catboost(self, booster):
+        catboost_params = get_catboost_params(booster)
+        self.daal_model_ = get_gbt_model_from_catboost(booster)
+        self._get_params_from_catboost(catboost_params)
+
+
+    def build_model(self, model):
+        (submodule_name, class_name) = (model.__class__.__module__,
+                                        model.__class__.__name__)
+        # Build from LightGBM
+        if (submodule_name, class_name) == ("lightgbm.basic", "Booster"):
+            self._build_model_from_lightgbm(model)
+        # Build from XGBoost
+        elif (submodule_name, class_name) == ("xgboost.core", "Booster"):
+            self._build_model_from_xgboost(model)
+        # Build from CatBoost
+        elif (submodule_name, class_name) == ("catboost.core", "CatBoost"):
+            self._build_model_from_catboost(model)
+        else:
+            raise TypeError(f"Unknown model format {submodule_name}.{class_name}")
+
+        self.is_regression_ = isinstance(self.daal_model_, gbt_regression_model)
+
+        return self
+
+    def _predict_classification(self, X, fptype, resultsToEvaluate):
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError('Shape of input is different from what was seen in `fit`')
+
+        if not hasattr(self, 'daal_model_'):
+            raise ValueError((
+                "The class {} instance does not have 'daal_model_' attribute set. "
+                "Call 'fit' with appropriate arguments before using this method.").format(
+                    type(self).__name__))
+
+        # Prediction
+        predict_algo = gbt_classification_prediction(
+            fptype=fptype,
+            nClasses=self.n_classes_,
+            resultsToEvaluate=resultsToEvaluate)
+        predict_result = predict_algo.compute(X, self.daal_model_)
+
+        if resultsToEvaluate == "computeClassLabels":
+            return predict_result.prediction.ravel().astype(np.int64, copy=False)
+        else:
+            return predict_result.probabilities
+
+    def _predict_regression(self, X, fptype):
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError('Shape of input is different from what was seen in `fit`')
+
+        if not hasattr(self, 'daal_model_'):
+            raise ValueError((
+                "The class {} instance does not have 'daal_model_' attribute set. "
+                "Call 'fit' with appropriate arguments before using this method.").format(
+                    type(self).__name__))
+
+        # Prediction
+        predict_algo = gbt_regression_prediction(fptype=fptype)
+        predict_result = predict_algo.compute(X, self.daal_model_)
+
+        return predict_result.prediction.ravel()
+
+    def predict(self, X, fptype="float", resultsToEvaluate="computeClassLabels"):
```

Comment:
Where is `predict_proba` and `predict_log_proba` methods?

Reply:
> Where is `predict_proba` and `predict_log_proba` methods?

This class is an analog of booster object from xgboost/lightgbm/catboost. It is not scikit-style estimator. Look [here](https://github.com/intel/scikit-learn-intelex/blob/3f34dcb5b2f97182f279d8bbbb5cb8eb36f2d298/daal4py/sklearn/ensemble/GBTDAAL.py#L191) for scikit-style classifier and regressor.

Reply:
From example:
```python
    # Conversion to daal4py
    daal_model = d4p.gbt_build_model(lgb_model)

    # daal4py prediction
    daal_prediction = daal_model.predict(X_test)
```
Is there a reason why `daal_model` should not be sklearn-like estimator with full set of prediction functions?

Reply:
The reason is that the frameworks (xgboost/lightgbm/catboost) have two parallel versions of API: functional and scikit-style. The problem is that one can't made a scikit-estimator from the Booster object from functional API for xgboost and lightgbm. We need to support model builders not only for scikit-style API users, but also for functional API users of the GBT Frameworks. GBTDAALModel is analog of the Booster object from functional API.

Reply:
> GBTDAALModel is analog of the Booster object from functional API

What is the need to mimic Booster object if most of it's methods are not implemented?

Reply:
> 

There is no any standard methods sets for Booster. Each framework has it's own set of methods. We need to support Booster-based API, as far as frameworks support it, and some users use it.